lib/elixir_bench/repos/repo.ex

```diff
@@ -16,4 +16,8 @@ defmodule ElixirBench.Repos.Repo do
     |> cast(attrs, [:owner, :name])
     |> validate_required([:owner, :name])
   end
+
+  def slug(%Repo{} = repo) do
```

Comment:
I wanted to ask if we need to match on repo here, but it's done in other parts of the code base as well :thinking: 

Upside I see is that it fails early when we don't pass a repo, downside is we can't pass a normal map.

Intuitively I wouldn't match on it but it also makes sense so :man_shrugging: 

Any input @michalmuskala ? ;P

Reply:
Yes, I also wanted to ask you when to do or not this kind of match.. in the ```ElixirBench.Benchmarks``` module, for example, some methods does this kind of match like in ```submit_job(%Job{} = job, results)``` but others don't ```def create_job(repo, attrs)```  

I think we can  remove the match indeed. This matches should be used in cases where different types make calls to the same method but with different behaviors, right?

Reply:
yeah I think we can remove it until michal tells us otherwise :)